### PR TITLE
wip

### PR DESCRIPTION
**Pull request by jakesmith**
## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [ ] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [ ] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [ ] I have read the CONTRIBUTORS document.
- [ ] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

### CODE REVIEW

File: common/remote/sockfile.cpp [hunk 1]
```diff
@@ -6339,6 +6339,8 @@ class CRemoteFileServer : implements IRemoteFileServer, public CInterface
             }
             else
             {
+                responseWriter->outputBeginArray("Msg");
```
Comment (afishbeck) [inline common/remote/sockfile.cpp]: You should actually begin a nested "Msg" object/item here, as well.  Each group of rows and each Eog is actually going to be inside a Msg object.

 responseWriter->outputBeginNested("Msg");

File: common/remote/sockfile.cpp [hunk 2]
```diff
@@ -6355,14 +6357,15 @@ class CRemoteFileServer : implements IRemoteFileServer, public CInterface
...
-                                responseWriter->outputBeginNested("Eog", false);
-                                responseWriter->outputEndNested("Eog");
                                 row = outputActivity->nextRow(outBuilder, rowSz);
                                 if (!row)
                                 {
                                     eoi = true;
                                     break;
                                 }
+                                responseWriter->outputEndArray("Row");
+                                responseWriter->outputBool(true, "Eog");
```
Comment (afishbeck) [inline common/remote/sockfile.cpp]: This Eog has to be inside a new message object, so:
responseWriter->outputEndArray("Row");
responseWriter->outputBeginNested("Msg");
responseWriter->outputBool(true, "Eog");
responseWriter->outputEndNested("Msg");
responseWriter->outputBeginArray("Row");

File: common/remote/sockfile.cpp [hunk 3]
```diff
@@ -6389,6 +6392,8 @@ class CRemoteFileServer : implements IRemoteFileServer, public CInterface
                         resultBuffer.clear();
                     }
                 }
+                responseWriter->outputEndArray("Row");
```
Comment (afishbeck) [inline common/remote/sockfile.cpp]: Also end the Msg object/item here.
responseWriter->outputEndNested("Msg");

**Comment by afishbeck** [comment]
@jakesmith some comments.

File: common/remote/sockfile.cpp (hunk 2 above)
Comment (jakesmith) [reply common/remote/sockfile.cpp]: I think needs to be.
                                responseWriter->outputEndArray("Row");
                                responseWriter->outputEndNested("Msg");
                                responseWriter->outputBeginNested("Msg", false);
                                responseWriter->outputBool(true, "Eog");
                                responseWriter->outputEndNested("Msg");
                                responseWriter->outputBeginNested("Msg", false);
                                responseWriter->outputBeginArray("Row");
but please check.

**Comment by jakesmith** [comment]
@afishbeck - added it a further commit.

But there is still the issue of the outer '{' '}' around the Response.
I can't see how you add that, please let me know, thanks.

**Comment by jakesmith** [comment]
Changed to cleaner form, embedding eog ("-Eog-") as a special field where eog marker required.

**Comment by jakesmith** [comment]
@afishbeck - please take a look

Also the outer object { } , I have a comment in testsocket:
                // JCSMORE - json string coming back from IXmlWriterExt is always rootless the moment, so workaround it by supplying ptr_noRoot to reader
                // writer should be fixed.

which probably applies. Can you take a look at the writer and see why it isn't creating a root object and if it can be changed without breaking backward compatibility with anything else that uses it?

**Comment by jakesmith** [comment]
@afishbeck - I'll turn this into an actual PR,
But need a fix for outer { } , which I think is a bug with the JSON impl. of IXmlWriterExt, as per comments above.

**Pull request closed without merging**